### ISSUE TITLE
double click is required first time using radiobutton and on onchange

### ISSUE THREAD

**Issue by Ammar-Khan-Attaullah** [dowjones/react-dropdown-tree-select#455]
using onchange method along  with radio button option is not getting selected. we have to click two times in order to select the option while selecting an option first time
while using radiobutton with onchange method.
if we click first time on any option it is not getting selected untill clicked twice

**Comment by mrchief** [comment]
This is not the behavior on the demo so I think something in your handler is interfering. If you can create a sandbox to reproduce this, we can dig further.

**Comment by sophia9211** [comment]
I have same problem
this is my code 

`import React, { FC, useEffect, useState } from "react";
import DropdownTreeSelect from "react-dropdown-tree-select";
import "react-dropdown-tree-select/dist/styles.css";
import "./DropdownContainer.scss";



const DropdownContainer = ({ data, getParentId }) => {
  const [treeData, setTreeData] = useState(data);

  useEffect(() => {
    setTreeData(data);
  }, [data]);

  const onChange = (e) => {
    getParentId(e.value);
  };

  return (
    <div className="dropdown_wrap">
      <DropdownTreeSelect data={treeData} mode="radioSelect" keepTreeOnSearch={false} onChange={onChange} />
    </div>
  );
};

export default DropdownContainer;
`

**Comment by Mako-L** [comment]
https://github.com/sanusart/react-dropdown-select
This might help.
Just needs a little change to become a tree select dropdown.

**Comment by aishwarya2392** [comment]
> using onchange method along with radio button option is not getting selected. we have to click two times in order to select the option while selecting an option first time
> while using radiobutton with onchange method.
> if we click first time on any option it is not getting selected untill clicked twice

I get this issue while clearing out the selected value

**Comment by mrchief** [comment]
![X3LwBqhpWS](https://user-images.githubusercontent.com/781818/110853370-8d6c2900-8281-11eb-8463-98f4805348d4.gif)

It works fine in the demo. Can anyone of you create a sandbox that reproduces this?

**Comment by Junikiewicz** [comment]
I had similar problem. I used react-hook-form and my onChange function was causing element rerender. Because of this I had to double-click on the option to see proper value in input field. Workaround mentioned in other similar issues helped: [prevent-re-render-on-parent-render-hoc approach.](https://dowjones.github.io/react-dropdown-tree-select/#/story/prevent-re-render-on-parent-render-hoc). Hope it helps.
